translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+    properties['start_date_accuracy'] = "estimated"
+
+    # end_date_accuracy
+    properties['end_date_accuracy'] = "estimated"
+
+    # beginning_accuracy
+    properties['beginning_accuracy'] = "estimated"
+
+    # ending_accuracy
+    properties['ending_accuracy'] = "estimated"
+
+    # road_name
+    road_name = alert['alert:RoadName']
+    properties['road_name'] = road_name
+
+    # direction
+    Direction_map = {'North': 'northbound', 'South': 'southbound', 'West': 'westbound', 'East': 'eastbound'}
+ 
+    properties['direction'] = Direction_map.get(alert['alert:Direction'])
+
+    # vehicle impact
+
+    properties['vehicle_impact'] = 'unknown' #get_vehicle_impact(alert['alert:ClosureType'])
+
+    # Relationship
+    properties['relationship'] = {}
+
+    # lanes
+    properties['lanes'] = []
+
+    # road_name
+    properties['road_number'] = ""
+
+    # beginning_cross_street
+    properties['beginning_cross_street'] = ""
+
+    # beginning_cross_street
+    properties['ending_cross_street'] = ""
+
+    # event status
+    properties['event_status'] = get_event_status(alert['alert:ReportedTime'], alert.get('alert:ExpectedEndTime'))
+
+    # type_of_work
+    # maintenance, minor-road-defect-repair, roadside-work, overhead-work, below-road-work, barrier-work, surface-work, painting, roadway-relocation, roadway-creation
+    properties['types_of_work'] = []
+
+    # reduced speed limit
+    properties['reduced_speed_limit'] = 25
+
+    # workers present
+    properties['workers_present'] = False
+
+    # restrictions
+    properties['restrictions'] = [] #will work with it late
+
+    # description
+    properties['description'] = alert['alert:Description']
+
+    # creation_date
+    properties['creation_date'] = reformat_datetime(alert['alert:ReportedTime'])
+
+    # update_date
+    
+    properties['update_date'] = reformat_datetime(alert['alert:LastUpdatedDate'])
+
+    feature = {}
+    feature['type'] = "Feature"
+    feature['properties'] = properties
+    feature['geometry'] = geometry
+
+    return feature
+
+#function to validate the alert
+def validate_alert(alert):
+    
+    if not alert or (type(alert) != dict and type(alert) != OrderedDict):
+        logging.warning('alert is empty or has invalid type')
+        return False
+    
+
+    polyline = alert.get('alert:Polyline')
+    coords = parse_polyline(polyline)
+    street = alert.get('alert:RoadName', '')
+
+    starttime = alert.get('alert:ReportedTime')
+    endtime = alert.get('alert:ExpectedEndTime')
+    description = alert.get('alert:Description')
+   
+    required_fields = [ polyline, coords, street, starttime, endtime, description]
+    for field in required_fields:
+        if not field:
+            logging.warning('Invalid alert with id =. Not all required fields are present')
+            return False
+            
+    try:
+        dateutil.parser.parse(alert['alert:ReportedTime'])
+        if alert.get('alert:ExpectedEndTime'):
```

Comment:
Same here, we know that ExpectedEndTime exists from above, and can reuse that `endtime` variable here.

Reply:
I changed that to end time